### ISSUE TITLE
Problems with assets 

### ISSUE THREAD

**Issue by evgenTraytyak** [shakacode/react_on_rails#401]
How should I use `svg` files or images in my components when i store them in `client/assets/images/`?

If I use path like this `/assets/some-image.png` in development env it works. But in production env this image has e-tag postfix `/assets/some-image-5dd021956efcd611492a65f2a05fb24a2c304d98ec00ecae3a7f17e48da10652.png` and it doesn't work because i have `/assets/some-image.png` path in my component .

How can I solve this problem without rails helpers `= assets_path()` and `.jsx.erb`?

I'm using `react_on_rails` v.4.

**Comment by evgenTraytyak** [comment]
@justin808 could you help me?

**Comment by justin808** [comment]
@evgenTraytyak are you trying to use webpack to build your CSS or Rails.

It's much simpler to just let Rails handle your CSS.

Otherwise, you can use CSS modules and follow the recipe here: https://github.com/shakacode/react-webpack-rails-tutorial/.

The reason you're seeing the problem with fingerprinting is that you're not using only the rails asset pipeline.

Please keep in mind that ReactOnRails has no direct connection with CSS assets, other than we support webpack and you can use webpack to build your CSS assets for Rails.

We should probably close this issue.

**Comment by ramusus** [comment]
@evgenTraytyak @justin808 I faced with the same problem. I need to use right images paths from jsx in `<img src="..." />`  and in javascript for src dynamic changing on the page. 

I used tutorial https://github.com/shakacode/react-webpack-rails-tutorial/ and everything works perfect in development until I deployed the code to heroku. We are using rails asset pipelines and all images was collected in `public/assets/image-5dd0219..........52.jpg`. I found it's required to convert original image name to the hashed name on production and added script 

```
// app/assets/javascripts/assets_images.js.erb
<%
imgs = {}
Dir.chdir("#{Rails.root}/app/assets/images/") do
  imgs = Dir.glob("**/*").inject({}) {|h,f| h.merge! f => asset_path(f)}
end
%>

window.imagePath = function(name) {
  return <%= imgs.to_json %>[name]
}
```
and required it in `app/assets/javascripts/application_non_webpack.js`
```
//= require assets_images
```
Then I start to use wrapper `imagePath('image.jpg')` everywhere and it make things works better: client rendering starts working as expected, but server rendering shows me error:

```
[Error] [SERVER] Exception in rendering!
	global code (xxx.herokuapp.com:117)
[Error] [SERVER] message: imagePath is not defined
	global code (xxx.herokuapp.com:118)
[Error] [SERVER] stack: ReferenceError: imagePath is not defined
    at RadiumEnhancer.render (<eval>:40131:58)
    at RadiumEnhancer.render (<eval>:36793:66)
    at ReactCompositeComponentMixin._renderValidatedComponentWithoutOwnerOrContext (<eval>:16344:35)
    at ReactCompositeComponentMixin._renderValidatedComponent (<eval>:16364:33)
    at ReactCompositeComponentMixin.mountComponent (<eval>:15977:31)
    at Object.ReactReconciler.mountComponent (<eval>:14295:36)
    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (<eval>:22839:45)
    at ReactDOMComponent.Mixin._createContentMarkup (<eval>:20022:33)
    at ReactDOMComponent.Mixin.mountComponent (<eval>:19910:30)
    at Object.ReactReconciler.mountComponent (<eval>:14295:36)
	global code (xxx.herokuapp.com:119)
```

I'm still searching solution how to make function `imagePath` accessible during server rendering. Perhaps register it somehow in `serverRegistration.jsx`? I tried, but without success, @justin808 do you have any suggestions?

**Comment by ramusus** [comment]
Solved this issue very simple - just put images to public folder and fingerprinting is not worrying me for assets images

**Comment by alleycat-at-git** [comment]
@evgenTraytyak @ramusus Have a look at [PR#413](https://github.com/shakacode/react_on_rails/pull/413) we're about to merge. I think it will solve your problems.

**Comment by robwise** [comment]
@justin808 suggest we close as @alleycat-at-git fixed this in #413
